### PR TITLE
ENH: add drop_fields method for tabular expressions

### PR DESCRIPTION
**Pull request by llllllllll**
simple helper for dropping fields of a table

### CODE REVIEW

**Comment by llllllllll** [comment]
ping @mtydykov

File: blaze/expr/expressions.py [hunk 1]
```diff
@@ -949,15 +955,60 @@ def ndim(expr):
...
 
 
+def drop_field(expr, field, *fields):
+    """Drop a field or fields from a tabular expression.
+
+    Parameters
+    ----------
+    expr : Expr
+        A tabular expression to drop columns from.
+    *fields
```
Comment (mtydykov) [inline blaze/expr/expressions.py]: `field` is missing from docstring

Comment (llllllllll) [inline blaze/expr/expressions.py]: This is intentional, this pattern is to make a *args which requires at least one argument.

File: blaze/expr/tests/test_expr.py [hunk 2]
```diff
@@ -331,3 +331,44 @@ def test_cast():
...
+        " (['a', 'b', 'c', 'd'])"
+    )
+
+    # test dropping a mix of correct and missing fields
+    with pytest.raises(ValueError) as e:
+        s.drop_field('a', 'fake')
+    assert (
+        str(e.value) ==
+        "fields ['fake'] were not in the fields of expr (['a', 'b', 'c', 'd'])"
+    )
```
Comment (mtydykov) [inline blaze/expr/tests/test_expr.py]: Should we add an assertion after each of these failed drops to ensure that the expr still has the fields we expect?

**Comment by mtydykov** [comment]
One last comment, but otherwise 👍

File: blaze/expr/tests/test_expr.py (hunk 2 above)
Comment (llllllllll) [inline blaze/expr/tests/test_expr.py]: expressions are immutable so this just returns the expression with less fields. The name is based on pandas `drop(axis=1)` which is not inplace by default.

**Pull request merged**